forgot_password/handlers.py

```diff
@@ -22,10 +27,66 @@
 from .util import email as emailutil
 from .util import user as userutil
 
-
 logger = logging.getLogger(__name__)
 
 
+class IllegalArgumentError(ValueError):
+    pass
+
+
+ResetPasswordRequestParams = namedtuple(
+    'ResetPasswordRequestParams',
+    ['code', 'user_id', 'expire_at', 'user', 'user_record'])
+
+
+def download_template(url, name):
+    path_prefix = 'templates/forgot_password'
+    path = '/'.join([path_prefix, name])
+
+    Path(path_prefix).mkdir(parents=True, exist_ok=True)
+    logger.info('Downloading {} from {}'.format(path, url))
+
+    try:
+        urlretrieve(url, path)
+    except HTTPError as e:
+        raise Exception('Failed to download {}: {}'.format(name, e.reason))
```

Comment:
> What will happen when this exception is thrown? Will the plugin runtime quit or will plugin fail to initialize? 

The plugin will still be initialized, since skygear-server would not care the result of lifecycle events.

> Include the URL in the exception so that the user knows which URL fails to load.

Okay.